dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/changelog/ChangeLogServiceTest.java

```diff
@@ -0,0 +1,311 @@
+    assertNoErrors(
+        trackerImportService.importTracker(
+            importParams, fromJson("tracker/event_and_enrollment.json")));
+  }
+
+  @Test
+  void shouldFailWhenEventDoesNotExist() {
+    assertThrows(
+        NotFoundException.class,
+        () -> changeLogService.getEventChangeLog(UID.of(CodeGenerator.generateUid())));
+  }
+
+  @Test
+  void shouldFailWhenEventOrgUnitIsNotAccessible() {
+    injectSecurityContextUser(manager.get(User.class, "o1HMTIzBGo7"));
+
+    assertThrows(
+        NotFoundException.class, () -> changeLogService.getEventChangeLog(UID.of("D9PbzJY8bJM")));
+  }
+
+  @Test
+  void shouldFailWhenEventProgramIsNotAccessible() {
+    injectSecurityContextUser(manager.get(User.class, "o1HMTIzBGo7"));
+
+    assertThrows(
+        NotFoundException.class, () -> changeLogService.getEventChangeLog(UID.of("G9PbzJY8bJG")));
+  }
+
+  @Test
+  void shouldFailWhenProgramTrackedEntityTypeIsNotAccessible() {
+    injectSecurityContextUser(manager.get(User.class, "FIgVWzUCkpw"));
+
+    assertThrows(
+        NotFoundException.class, () -> changeLogService.getEventChangeLog(UID.of("H0PbzJY8bJG")));
+  }
+
+  @Test
+  void shouldFailWhenProgramWithoutRegistrationAndNoAccessToEventOrgUnit() {
+    injectSecurityContextUser(manager.get(User.class, "o1HMTIzBGo7"));
+
+    assertThrows(
+        NotFoundException.class, () -> changeLogService.getEventChangeLog(UID.of("G9PbzJY8bJG")));
+  }
+
+  @Test
+  void shouldReturnChangeLogsWhenDataValueIsCreated() throws NotFoundException {
+    injectSecurityContextUser(manager.get(User.class, "M5zQapPyTZI"));
+
+    Event event = manager.get(Event.class, "QRYjLTiJTrA");
+    assertNotNull(event);
+    String dataElementUid = event.getEventDataValues().iterator().next().getDataElement();
+    DataElement dataElement = manager.get(DataElement.class, dataElementUid);
+    assertNotNull(dataElement);
+
+    List<EventChangeLog> changeLogs = changeLogService.getEventChangeLog(UID.of("QRYjLTiJTrA"));
+
+    assertAll(
+        () -> {
+          validateNonValueElements(1, changeLogs, importUser, dataElementUid);
+          assertEquals("15", changeLogs.get(0).change().dataValue().currentValue());
+          assertNull(changeLogs.get(0).change().dataValue().previousValue());
+        });
+  }
+
+  @Test
+  void shouldReturnChangeLogsWhenDataValueIsDeleted() throws NotFoundException, IOException {
+    injectSecurityContextUser(manager.get(User.class, "M5zQapPyTZI"));
+
+    Event event = manager.get(Event.class, "QRYjLTiJTrA");
+    assertNotNull(event);
+    String dataElementUid = event.getEventDataValues().iterator().next().getDataElement();
+    DataElement dataElement = manager.get(DataElement.class, dataElementUid);
+    assertNotNull(dataElement);
+
+    TrackerObjects trackerObjects = fromJson("tracker/event_and_enrollment.json");
+    updateDataValue(trackerObjects, event.getUid(), dataElementUid, "");
```

Comment:
move 

    TrackerObjects trackerObjects = fromJson("tracker/event_and_enrollment.json");
    assertNoErrors(trackerImportService.importTracker(importParams, trackerObjects));

into `updateDataValue`

It will simplify its signature and make the test even nicer to read